tests/components/template/test_lock.py

```diff
@@ -14,13 +20,15 @@
         "data_template": {
             "action": "lock",
             "caller": "{{ this.entity_id }}",
+            "code": "{{ code }}",
```

Comment:
Why is this change required?

Reply:
Because tests fail otherwise 🤷‍♂️ 😉 

I chimed into the way the other tests in this file use this config preset to assess whether actions are called with correct parameters.

Reply:
That's not an answer.

The code is needed only by your newly added tests. By reverting your changes in the test file, I can run all tests successfully.

Reply:
Seriously? If "I chimed into the way the other tests in this file use this config preset to assess whether actions are called with correct parameters." is not an answer then I don't know what is.

I guess you know how the tests work in this file, so what's the matter here? I honestly don't get it 🤷

Reply:
Providing a `code` is optional for the lock service calls.
I would leave the current service as they are (meaning I would revert this change)
If your newly added tests need a code, they add it only for these particular tests.
Could you also add a test in which a code_format is configured but no code is passed on the service call?

Reply:
I agree with @edenhaus, add a new config dict `OPTIMISTIC_LOCK_CONFIG_WITH_CODE` and use that in the new tests + add the negative test he requested.

Once that's sorted, this PR is good to go 👍

Reply:
That test that @edenhaus mentioned is already there; however I made that section a little more complete. Specific lock config for new tests is there as well.